header.php

```diff
@@ -1,588 +1,593 @@
+		} //$PASSWORD != $RE_TYPE_PASSWORD
+		else if (strlen($PASSWORD) < '8') {
+        $MESSAGE_REGISTER= "Password Must Contain At Least 8 Characters!";
+    	}
+		else if (!filter_var($E_MAIL, FILTER_VALIDATE_EMAIL)) {
+  $MESSAGE_REGISTER = "Invalid Email";
+		}
+		
+		// enter user to database
+		else {
+				$insert_user        = "INSERT INTO `tblusers`(`Username`, `FirstName`, `LastName`, `EmailAddress`, `PhoneNumber`, `Password`, `SecurityQuestion`, `SecurityAnswer`) VALUES ('$USERNAME','$FIRST_NAME','$LAST_NAME','$E_MAIL','$PHONE_NUMBER','" . md5($PASSWORD) . "','$SECURITY_QUESTION','$SECURITY_ANSWER')";
+				$result_insert_user = mysqli_query($con, $insert_user);
+				if ($result_insert_user) {
+						$MESSAGE_REGISTER = "successful";
+				} //$result_insert_user
+				else {
+						$MESSAGE_REGISTER = "Registration unsuccessful!";
+				}
+		}
+		mysqli_close($con); // Closing connection
+} //isset($_POST["register"])
+?>
+
+<?php
+///////////////////////////////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////View Products page related .php////////////////////////////////////
+///////////////////////////////////////////////////////////////////////////////////////////////////////
+require('db.php');
+///////////////////////Add products to cart///////////////////////
+$MESSAGE_REVIEW = '';
+if (isset($_POST['addtocart'])) {
+		if (!isset($_SESSION['username'])) {
+				$_SESSION['selectedproduct']  = $_POST['productid'];
+				$_SESSION['selectedquantity'] = $_POST['quantity'];
+				$_SESSION['crumbs']           = "1";
+				header("location: login.php");
+		} //!isset($_SESSION['username'])
+		else {
+				$PRODUCT_ID              = $_POST['productid'];
+				$PRODUCT_QUANTITY        = $_POST['quantity'];
+				$USERNAME                = $_SESSION['username'];
+				$v_insert_to_cart        = "INSERT into`tblcart` (ProductID, Username, Quantity) VALUES ('" . $PRODUCT_ID . "','" . $USERNAME . "','" . $PRODUCT_QUANTITY . "')";
+				$result_v_insert_to_cart = mysqli_query($con, $v_insert_to_cart);
+				header('location: cart.php');
+		}
+} //isset($_POST['addtocart'])
+///////////////////////Add a review///////////////////////
+if (isset($_POST['addreview'])) {
+	if (!isset($_POST['productid']) || !isset($_POST['review'])) {
+		$MESSAGE_REVIEW = "Unable to Place Review!";
+	} else {
+		$PRODUCT_ID = $_POST['productid'];
+		$USERNAME = $_SESSION['username'];
+		$PRODUCT_REVIEW = $_POST['review'];
+
+		$insert_review = "INSERT INTO `tblproductreviews`(`ProductID`, `Username`, `Review`) VALUES (?, ?, ?)";
+		$stmt = mysqli_prepare($con, $insert_review);
+		mysqli_stmt_bind_param($stmt, 'iss', $PRODUCT_ID, $USERNAME, $PRODUCT_REVIEW);
+
+		if (mysqli_stmt_execute($stmt)) {
+			$MESSAGE_REVIEW = "Thank you for your review!";
+			header('location: #.php');
+		} else {
+			$MESSAGE_REVIEW = "Unable to Place Review!";
+			header('location: #.php');
+		}
+
+		mysqli_stmt_close($stmt);
+	}
+}

```

Comment:
Use prepared statements with parameter binding to prevent `SQL` injection.Removed unnecessary concatenation in the SQL query and used placeholders instead.Used `mysqli_stmt_execute()` to execute the prepared statement. Closed the prepared statement `using  mysqli_stmt_close()`